src/SmartComponents/SystemsTable/SystemsTable.js

```diff
@@ -93,10 +93,14 @@ class SystemsTable extends React.Component {
     }
 
     systemFetch = () => {
-        const { client } = this.props;
+        const { client, showOnlySystemsWithTestResults } = this.props;
         const { policyId, perPage, page, activeFilters } = this.state;
         let filter = this.filterBuilder.buildFilterString(activeFilters);
 
+        if (showOnlySystemsWithTestResults) {
+            filter = `has_test_results = true and ${filter}`;
```

Comment:
Do we need parentheses? Imagine the case where the filter is currently `FOO or BAR`, then this would result in `has_test_results = true and FOO or BAR`, interpreted as `(has_test_results = true and FOO) or BAR` instead of `has_test_results = true and (FOO or BAR)`

```suggestion
            filter = `has_test_results = true and (${filter})`;
```